gsy_myco_sdk/matchers/redis_base_matcher.py

```diff
@@ -23,9 +23,13 @@ def __init__(self, redis_url="redis://localhost:6379",
         self.redis_db = StrictRedis.from_url(redis_url)
         self.pubsub = self.redis_db.pubsub() if pubsub_thread is None else pubsub_thread
         self.executor = ThreadPoolExecutor(max_workers=MAX_WORKER_THREADS)
+        self._connect_to_simulation()
```

Comment:
I think it would be good to use a similar interface as here: https://github.com/gridsingularity/gsy-myco-sdk/blob/d0032ff86f2a731e9c8d3bff32c14828ad867ef6/gsy_myco_sdk/matchers/base_matcher.py#L28-L29

What do you think? Or should the redis matcher always connect immediately?

Reply:
IMO this extra keyword does not make sense, because the matcher is useless without a connection to a simulation. I'd rather remove the `auto_connect` parameter from the interface that you mentioned actually. What do you think?

Reply:
I'm ok with what you propose. I'd just like to have @Hassan754's opinion to know if there was a specific reason to add it.

Reply:
The initializer of the `RedisBaseMatcher` is slightly different than the rest base matcher, in this matcher we don't have the `auto_connect` argument so the proposal is not valid here, because it's true that it should always connect to Redis

Reply:
@Hassan754 ok but my question was about @hannesdiedrich's proposal:
> I'd rather remove the auto_connect parameter from the interface that you mentioned actually. What do you think?

Reply:
@fievelk This is my point, the auto_connect isn't in the interface `MycoMatcherClientInterface` nor in the `RedisBaseMatcher` (where the comment originated from), it's only in the initializer of the Rest matcher, and it makes sense there

Reply:
Sorry. I wasn't talking strictly about the shared interface class (`MycoMatcherClientInterface`). I was talking more broadly about the "interface" of the `BaseMatcher` vs `RedisBaseMatcher`. Given their names, I was thinking that they would have had the same interface and usage. In that sense, the fact that one of them has `auto_connect` (and the other doesn't) seemed a bit strange to me.
So the main question is basically this: why does `auto_connect` make sense in the REST `BaseMatcher`, but not in the `RedisBaseMatcher`? Can you help me understand this?

Reply:
I don't know I understand your worry about the fact that two classes differ in the initialization, this seems normal to my eye, the `MycoMatcherClientInterface` is an interface (with the spirit of an abstract class) so we're defining the common endpoints that should be shared between them (not ALL interfaces should be homogenized though unless we're dealing with actual inheritance )
Anyway, regarding the `auto_connect`, I originally added this to allow the extending class to perform actions before actually connecting to the websocket, if you believe this is not realistic, feel free to remove it

Reply:
@hannesdiedrich FYI. 
My idea is that for the users it would be easier to have the same interface: using REST or Redis should be more or less the same for them, without differences in the way classes are initialised. In this way, they could re-use the same code and just change the class name. So I would personally either add this parameter to both or remove it from both. But if you don't see the need for it, proceed as you think it's best.

Reply:
Ok, from this long discussion I take:
> I originally added this to allow the extending class to perform actions before actually connecting to the websocket

and 
> My idea is that for the users it would be easier to have the same interface: using REST or Redis

which results in adding the `auto_connect` flag to the RedisBaseMatcher. Will so that.

Reply:
Sorry for dragging this thread further, but for a newcomer to the discussion, why do we need the auto_connect in the REST case ? It could have been mentioned already but I missed it through the message thread, apologies in advance in this case. 
In general though, looking at the constructor of the `BaseMatcher` , I see no particular reason for having this parameter and the case looks similar to the Redis one, only instead of Redis we have a Websocket connection.

Reply:
Ok, now we have two main points: 
> My idea is that for the users it would be easier to have the same interface: using REST or Redis

and
> why do we need the auto_connect in the REST case?

As I was in favour of removing the `auto_connect` at the beginning of this discussion, I will remove it now.